model/src/main/kotlin/Provenance.kt

```diff
@@ -79,6 +87,13 @@ data class RepositoryProvenance(
      * Return true if this provenance matches the processed VCS information of the [package][pkg].
      */
     override fun matches(pkg: Package): Boolean = vcsInfo == pkg.vcsProcessed
+
+    /**
+     * Return true if the [Provenance][other] is a [RepositoryProvenance] and its normalized vcsInfo matches this
+     * provenance's normalized [VcsInfo][vcsInfo].
+     */
+    override fun matches(other: Provenance): Boolean =
+        other is RepositoryProvenance && other.vcsInfo.normalize().matches(vcsInfo.normalize())
```

Comment:
Why is this disregarding the `resolvedRevision`? And if that's a mistake, why not simply use `provenanveA == provenanceB` for matching `Provenance`s?

Reply:
It was not done on purpose.

> why not simply use `provenanveA == provenanceB` for matching `Provenance`s?

Because `Repository` used the `equalsDisregardingPath` style `matches` before, and I wanted to ensure the same behavior as before. IIRC some tests would also fail with just `==` between `VcsInfo`, which `provenanveA == provenanceB` would imply, correct?

Reply:
> which `provenanveA == provenanceB` would imply, correct?

Yes, `==` on `data class`es is a "recursive equals()".

Reply:
Added a comparison of `resolvedRevision`.